plumbum/machines/session.py

```diff
@@ -245,6 +246,13 @@ def __del__(self):
         except Exception:
             pass
 
+    @property
+    def startup_result(self):
```

Comment:
What's the use for this? Would a bool property "started" or something like that be better and less tied to the implementation?

Reply:
The stderr saved in `ShellSession().startup_result` gets parsed by `SshTunnel().__init__()` to get the port number than OpenSSH has bound on the remote end of a reverse tunnel.

For example, for `SshMachine().tunnel(1234, 0, reverse=True)`, the underlying `ssh -R 0:localhost:1234 ...` call will have OpenSSH allocate a free remote port and report it on the stderr of the `ssh -R` command. This command is executed by `self.run("")` during `ShellSession.__init__`. The stderr needs to be saved so that `SshTunnel().__init__()` can parse it to get the bound port number.

I agree that this is tied to the implementation in a confusing way, but I don't think that `ssh -R ...`'s stderr can be captured other than by saving the return value of that initial `ShellSession().run("")`. The alternative would be to have `SshTunnel().__init__()` or `SshMachine().tunnel()` run the OpenSSH command that initializes the tunneled session, but that would break `ShellSession`s that aren't created by `SshMachine().tunnel()`.

Reply:
Okay, I'd think you could do the parsing here and store the interesting information rather than the raw stderr. I'd assume this at least is not valid for Paramiko machine?

Reply:
We could have `ShellSession().__init__` do the parsing and save a `reverse_tunnel_detected_rport` property rather than the current `startup_result` property. I didn't do that initially since I think it pushes the port detection logic even further away from `SshMachine().tunnel` and is more convoluted, but I can move it if you prefer.

Reply:
Would it be okay to leave this with hidden only access for now? `_startup_result`? Just to give us the option to do this in the future?

Reply:
Yeah! Definitely makes sense to not make it public API. Addressed with a fixup: https://github.com/tomerfiliba/plumbum/compare/fd767673b90f48682369a307b10fd1cbcd92a5e9..f55d299b335ba3d4842d7da24593d08a943c0795